WordPressKit/BlogServiceRemoteREST.m

```diff
@@ -287,6 +287,25 @@ - (void)fetchSiteInfoForAddress:(NSString *)siteAddress
                           }];
 }
 
+- (void)fetchUnauthenticatedSiteInfoForAddress:(NSString *)siteAddress
+                        success:(void(^)(NSDictionary *siteInfoDict))success
+                        failure:(void (^)(NSError *error))failure
+{
+    NSString *path = [self pathForEndpoint:@"connect/site-info" withVersion:ServiceRemoteWordPressComRESTApiVersion_1_1];
+    NSURL *siteURL = [NSURL URLWithString:siteAddress];
+
+    [self.wordPressComRestApi GET:path
+                       parameters:@{ @"url": siteURL.absoluteString }
+                          success:^(id responseObject, NSHTTPURLResponse *httpResponse) {
+                              if (success) {
+                                  success((NSDictionary *)responseObject);
+                                  return;
+                              }
+                          } failure:^(NSError *error, NSHTTPURLResponse *httpResponse) {
+                              failure(error);
```

Comment:
The documentation in the header says `failure` can be `nil`.  Should we wrap the call in an `if` block to check like we do for `success`?

Reply:
👍